src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,655 @@ const ActorInit En_Rz_InitVars = {
+    f32 temp_fa0;
+    f32 temp_fa1;
+
+    path = this->path;
+    if (path == NULL) {
+        return true;
+    }
+
+    pointPos = &((Vec3s*)Lib_SegmentedToVirtual(path->points))[this->unk_3BC];
+    temp_fa0 = pointPos->x - this->actor.world.pos.x;
+    temp_fa1 = pointPos->z - this->actor.world.pos.z;
+    this->actor.world.rot.y = Math_Atan2S(temp_fa0, temp_fa1);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 2, 2000, 200);
+
+    if ((SQ(temp_fa0) + SQ(temp_fa1)) < SQ(10.0f)) {
+        if (this->unk_420 & 1) {
+            this->unk_3BC--;
+            if (this->unk_3BC < 0) {
+                return 2;
+            }
+        } else {
+            this->unk_3BC++;
+            if (this->unk_3BC >= path->count) {
+                return 2;
+            }
+        }
+        return true;
+    }
+    return false;
+}
+
+s32 EnRz_CanTalk(EnRz* this, PlayState* play) {
+    if (this->actor.xzDistToPlayer < 80.0f && (fabsf(this->actor.playerHeightRel) < 20.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80BFBCEC(EnRz* this, PlayState* play) {
+    s32 pad;
+    Player* player = GET_PLAYER(play);
+
+    if (this->actor.xzDistToPlayer < 100.0f && Actor_IsFacingPlayer(&this->actor, 0x3000) &&
+        player->stateFlags2 & PLAYER_STATE2_2000000) {
+        return true;
+    }
+
+    return false;
+}
+
+void EnRz_UpdateAnimation(EnRz* this, PlayState* play) {
+    if (this->currentAnimIndex < 7) {
+        SkelAnime_Update(&this->skelAnime);
+    } else {
+        LinkAnimation_Update(play, &this->skelAnime);
+    }
+}
+
+Actor* func_80BFBDA0(EnRz* this, PlayState* play) {
+    Actor* npc = play->actorCtx.actorLists[ACTORCAT_NPC].first;
+
+    while (npc != NULL) {
+        if ((npc->id == ACTOR_EN_RZ) && (EN_RZ_GET_PARAM_F(&this->actor) == EN_RZ_GET_PARAM_F(npc))) {
+            if (&this->actor != npc) {
+                return npc;
+            }
+        }
+        npc = npc->next;
+    }
+    return NULL;
+}
+
+void func_80BFBDFC(PlayState* play) {
+    if ((gSaveContext.save.weekEventReg[75] & 0x80)) {
+        func_80151BB4(play, 0x27);
+    }
+    func_80151BB4(play, 0xC);
+}
+
+void EnRz_Destroy(Actor* thisx, PlayState* play) {
+    EnRz* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+s32 func_80BFBE70(EnRz* this, PlayState* play) {
+    u16 action;
+
+    if (!EN_RZ_GET_PARAM_8000(&this->actor) && this->currentAnimIndex == 4) {
+        func_800B9010(&this->actor, NA_SE_EV_CLAPPING_2P - SFX_FLAG);
+    }
+    if (Cutscene_CheckActorAction(play, this->csAction)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csAction));
+        action = play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csAction)]->action;
+        if (this->unk_428 != action) {
+            this->unk_428 = action;
+            switch (action & 0xFFFF) {
```

Comment:
```suggestion
            switch (action) {
```
Is the 0xFFFF needed? since `action` is `u16` ido might put the `& 0xFFFF` automatically.

Reply:
![image](https://user-images.githubusercontent.com/96613413/185785429-0392f2a2-c01a-458d-8dcf-e89630c4a96c.png)

Reply:
```suggestion
            switch (this->actionIndex) {
```
matches: https://decomp.me/scratch/Tn9v8